database/raft/raft.go

```diff
@@ -627,6 +654,15 @@ func (sv *Service) join(addr, baseURL string) error {
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		errmsg, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			return errors.Wrap(err, "could not parse response from boot server")
+		}
+		defer resp.Body.Close()
+		return errors.New(fmt.Sprintf("boot server responded with status %d: %s", resp.StatusCode, errmsg))
```

Comment:
_omg_ I couldn't remember what that function was, I spent about a minute staring at my screen trying to remember. 😳